plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java

```diff
@@ -202,25 +208,50 @@ public static void waitForProjects(Runnable delayTactic, IProject... projects) {
         Collections.addAll(jobs, jobManager.find("org.eclipse.wst.server.core.family"));
         Collections.addAll(jobs, jobManager.find("org.eclipse.wst.server.ui.family"));
         Collections.addAll(jobs, jobManager.find(ValidationBuilder.FAMILY_VALIDATION_JOB));
+
+        // Remove SLEEPING jobs as they may not run for a long time.
+        // May be specific to FAMILY_AUTO_BUILD jobs.
+        for (Iterator<Job> iter = jobs.iterator(); iter.hasNext();) {
+          Job job = iter.next();
+          if (job.getState() != Job.RUNNING && job.getState() != Job.WAITING) {
+            iter.remove();
+          }
+        }
+
+        // track whether we've reached a fixpoint in the build errors
         allBuildErrors = getAllBuildErrors(projects);
-        if (DEBUG) {
+        buildErrorsChanging =
+            numberOfPreviousBuildErrors < 0 || numberOfPreviousBuildErrors != allBuildErrors.size();
+        numberOfPreviousBuildErrors = allBuildErrors.size();
+
+        if (DEBUG || timer.elapsed(TimeUnit.SECONDS) > 10) {
+          System.err.printf("ProjectUtils#waitForProjects: waiting %s\n", timer);
           if (!jobs.isEmpty()) {
             System.err.printf("ProjectUtils#waitForProjects: waiting for %d jobs: %s\n",
                 jobs.size(), jobs);
-          } else if (!allBuildErrors.isEmpty()) {
+          }
+          if (!allBuildErrors.isEmpty()) {
             System.err.printf("ProjectUtils#waitForProjects: waiting for %d build errors\n",
                 allBuildErrors.size());
-          } else {
+            // don't report unchanged errors
+            if (buildErrorsChanging) {
+              for (String buildError : allBuildErrors) {
+                System.err.printf("  error: " + buildError);
+              }
+            }
+          }
+          if (timer.elapsed(TimeUnit.SECONDS) > 60) {
```

Comment:
So, if we go over 60 seconds, I guess we will print the state very frequently, every time we loop? A small interval might be a good idea.